lite2/verifier.go

```diff
@@ -16,6 +16,15 @@ var (
 	DefaultTrustLevel = tmmath.Fraction{Numerator: 1, Denominator: 3}
 )
 
+// Verify verifies the new header (h2) against the old header (h1). It ensures that:
+//
+//	a) h1 can still be trusted (if not, ErrOldHeaderExpired is returned);
+//	b) h2 is valid;
+//	c) either h2.ValidatorsHash equals h1NextVals.Hash()
+//		 OR trustLevel ([1/3, 1]) of last trusted validators (h1NextVals) signed
+//		 correctly  (if not, ErrNotEnoughVotingPowerSigned is returned);
```

Comment:
IMO there should be two different errors for the two cases:
- the validator set of a skipped header cannot be trusted, i.e. <1/3rd of h1 validator set has signed (new error, something like `ErrValidatorSetCannotBeTrusted`)
- the validator set is trusted but < 2/3rds has signed (`ErrNotEnoughVotingPowerSigned`)

Reply:
I agree; I think errors should be as specific as possible, if we are checking a different conditional it should be a different error.